segments-explainer/SEGMENTS-EXPLAINER.md

```diff
@@ -0,0 +1,105 @@
+
+A user may at any point take the browser window out of spanning mode and place it on one of the screens or vice-versa, in those cases the window resize event will fire and authors can query and get the number of available viewport segments.
+
+This proposal doesn't aim to substitute existing APIs &mdash; the proposed development model can be summarized as requesting current viewport segments on interesting events and adjusting to the new presentation environment. There are no additional lifecycle proposals - the viewport segments are immutable and developers would request them upon common sense events (e.g. orientationchange, resize). It also  doesn't suggest how developers would use viewport segments to position, scale and orient content - in practical explorations developers used viewport segments to select the best declarative layout, not to modify layouts in script, but either would be possible.
+
+## Security and Privacy
+
+### APIs availability in iframe context
+
+`segments` will return null when called from within an `iframe` context.
+
+## Examples of user experiences and solution outlines that can leverage two screens:
+
+Let's take a look at a few practical examples of the scenarios above and how viewport segments would allow to resolve them for better user experience. In each case we'll start with some existing scenario and complicate it to provide opportunity to apply the proposal.
+
+### A map application that presents a map on one window segment and search results on another
+
+![Foldable with the left segment of the window containing a map and the right segment containing list of search results](map-app.svg)
+
+#### JavaScript solution outline:
+
+```js  
+const segments = window.visualViewport.segments;
+
+if (segments && segments.length > 1) {
+	// now we know the device is a foldable
+	// and we can update CSS classes in our layout as appropriate 
+	document.body.classList.add('is-foldable');
+	document.querySelector('.map').classList.add('flex-one-half');
+	document.querySelector('.locations-list').classList.add('flex-one-half');
+}
+```
+
+We can also use the [@horizontal-viewport-segments](https://drafts.csswg.org/mediaqueries-5/#mf-horizontal-viewport-segments) and [@vertical-viewport-segments](https://drafts.csswg.org/mediaqueries-5/#mf-vertical-viewport-segments) media queries to change the layout and style of a website depending on the way a device is oriented. 
+
+Additionally, there are a number of new [environment variables](https://drafts.csswg.org/css-env-1/#viewport-segments) that can be used to place and position content on dual screen devices. 
+
+
+### Reacting to map application resize/spanning state change
+
+![Foldable with the left segment of the window containing browser and location finder website, right segment containing calculator app](map-app-resized.svg)
+
+#### JavaScript solution outline:
+
+```js  
+window.onresize = function() {
+	const segments = window.visualViewport.segments;
```

Comment:
Sorry for missing this before, but this needs the same treatment as the previous script block. Also, we probably want to have something like:

```js  
window.onresize = function() {
    const segments = window.visualViewport.segments;
    if (segments && segments.length > 1) {
        // Make changes two split content into the segments.
    } else {
        // Reset state to single viewport (normal responsive layout).
    }
}
```